Bug: slice index out of bounds when streaming logs

During my recent push to start dogfooding Brigade 2, I encountered this issue in the API server:

```
2021/06/28 19:18:54 Starting Brigade API Server -- version v2.0.0-alpha.5.1 -- commit 2875689
2021/06/28 19:18:54 API server is listening with TLS enabled on 0.0.0.0:8080
panic: runtime error: slice bounds out of range [:-1]

goroutine 307 [running]:
github.com/brigadecore/brigade/v2/apiserver/internal/core/kubernetes.(*logsStore).StreamLogs.func2(0xc000806620, 0xc000273380, 0x1d46440, 0xc000734f90)
	/src/v2/apiserver/internal/core/kubernetes/logs_store.go:122 +0x59a
created by github.com/brigadecore/brigade/v2/apiserver/internal/core/kubernetes.(*logsStore).StreamLogs
	/src/v2/apiserver/internal/core/kubernetes/logs_store.go:110 +0x350
```

This cause the API server to restart repeatedly and eventually enter a crash loop. 

In the end, every component does what it is was designed to do and the intermittent unavailability of the API server just slowed everything down an awful lot.

I've never experienced this issue previously, but it cropped up repeatedly in my first attempts at dogfooding.

Fix is already in the works.

--- a/v2/apiserver/internal/core/kubernetes/logs_store.go
+++ b/v2/apiserver/internal/core/kubernetes/logs_store.go
@@ -117,6 +117,9 @@ func (l *logsStore) StreamLogs(
 			if err == io.EOF {
 				break
 			}
+			if len(logLine) == 0 {
+				continue
+			}
 			// The last character should be a newline that we don't want, so let's
 			// remove that
 			logLine = logLine[:len(logLine)-1]

